src/nccl_ofi_net.c

```diff
@@ -943,19 +925,28 @@ static ncclResult_t ofi_connect(int dev, void *handle, void **sendComm)
 
 	req = allocate_nccl_ofi_request(sComm->nccl_ofi_reqs_fl);
 	if (OFI_UNLIKELY(req == NULL)) {
-			ret = ncclSystemError;
-			NCCL_OFI_WARN("Unable to get NCCL OFI request for device %d",
-				      sComm->dev);
-			goto error;
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("Unable to get NCCL OFI request for device %d",
+			      sComm->dev);
+		goto error;
+	}
+
+	sComm->local_ep_name = calloc(1, MAX_EP_ADDR);
+	if (OFI_UNLIKELY(sComm->local_ep_name == NULL)) {
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("Couldn't allocate sendComm for local EP name");
+		goto error;
 	}
+	memcpy(sComm->local_ep_name, remote_ep_addr, MAX_EP_ADDR);
 
 	req->sComm = sComm;
 	req->dev = sComm->dev;
 	req->direction = NCCL_OFI_SEND;
 
 	/* Send "connect" message to remote EP */
 	do {
-		rc = fi_tsend(sComm->local_ep, NULL, 0, NULL, sComm->remote_ep,
+		rc = fi_tsend(sComm->local_ep, sComm->local_ep_name,
```

Comment:
This to me implies that you are sending the same data (`remote_ep_addr`) we received from handle back to the source. What is the reason behind it?

If this is to identify different connections, then we use tags for that purpose. It looks like a redundant information to me.